dream_layer_backend/dream_layer_backend_utils/api_key_injector.py

```
@@ -46,102 +50,100 @@
+            needed_env_keys.add('IDEOGRAM_API_KEY')
+            logger.debug(f"Found {class_type} node - needs {needed_env_keys}")
+        elif 'Luma' in class_type:
+            needed_env_keys.add('LUMA_API_KEY')
+            logger.debug(f"Found {class_type} node - needs {needed_env_keys}")
     
-    # Add the chosen key to extra_data
-    if api_key_comfy_org:
-        workflow_with_keys["extra_data"]["api_key_comfy_org"] = api_key_comfy_org
-        print(f"[DEBUG] Injected api_key_comfy_org into workflow")
+    logger.debug(f"needed_env_keys: {needed_env_keys}")
+    logger.debug(f"all_api_keys keys: {all_api_keys.keys()}")
```

Ruff (0.12.2)

Undefined name `all_api_keys`

(F821)